mp/src/game/server/momentum/mom_player.cpp

```diff
@@ -2116,18 +2144,60 @@ void CMomentumPlayer::DoPaint()
     m_flNextPaintTime = gpGlobals->curtime + PAINT_CYCLE_TIME;
 }
 
-CON_COMMAND(toggle_duck, "Toggles duck state of the player. Only usable in the Ahop gamemode!\n")
+CON_COMMAND(toggle_duck, "Toggles duck state of the player.\n")
 {
-    if (!g_pGameModeSystem->GameModeIs(GAMEMODE_AHOP))
+    const auto pPlayer = CMomentumPlayer::GetLocalPlayer();
+    if (!pPlayer || pPlayer->GetFlags() & FL_FROZEN)
         return;
 
-    const auto pPlayer = UTIL_GetCommandClient();
-    if (!pPlayer)
+    if (pPlayer->m_iButtonsToggled & IN_DUCK)
+        pPlayer->m_iButtonsToggled &= ~IN_DUCK;
+    else
+        pPlayer->m_iButtonsToggled |= IN_DUCK;
+}
+
+CON_COMMAND(toggle_jump, "Toggles jump state of the player.\n")
+{
+    // no point if gamemode doesn't have autohop
+    if (!g_pGameModeSystem->GetGameMode()->PlayerHasAutoBhop())
+        return;
+
+    const auto pPlayer = CMomentumPlayer::GetLocalPlayer();
+    if (!pPlayer || pPlayer->GetFlags() & FL_FROZEN)
         return;
 
-    if (pPlayer->GetFlags() & FL_FROZEN)
+    if (pPlayer->m_iButtonsToggled & IN_JUMP)
+        pPlayer->m_iButtonsToggled &= ~IN_JUMP;
+    else
+        pPlayer->m_iButtonsToggled |= IN_JUMP;
+}
+
+CON_COMMAND(toggle_speed, "Toggles sprint state of the player. Toggle resets when unable to sprint.\n")
+{
+    if (!g_pGameModeSystem->GetGameMode()->HasCapability(GameModeHUDCapability_t::CAP_HUD_KEYPRESS_SPRINT))
         return;
 
-    pPlayer->ToggleDuck();
+    const auto pPlayer = CMomentumPlayer::GetLocalPlayer();
+    if (!pPlayer || pPlayer->GetFlags() & FL_FROZEN)
+        return;
+
+    if (pPlayer->m_iButtonsToggled & IN_SPEED)
+        pPlayer->m_iButtonsToggled &= ~IN_SPEED;
+    else
+        pPlayer->m_iButtonsToggled |= IN_SPEED;
 }
 
+CON_COMMAND(toggle_walk, "Toggles walk state of the player. Toggle resets when unable to walk.\n")
+{
+    if (!g_pGameModeSystem->GetGameMode()->HasCapability(GameModeHUDCapability_t::CAP_HUD_KEYPRESS_WALK))
+        return;
+
+    const auto pPlayer = CMomentumPlayer::GetLocalPlayer();
+    if (!pPlayer || pPlayer->GetFlags() & FL_FROZEN)
+        return;
+
+    if (pPlayer->m_iButtonsToggled & IN_WALK)
+        pPlayer->m_iButtonsToggled &= ~IN_WALK;
+    else
+        pPlayer->m_iButtonsToggled |= IN_WALK;
```

Comment:
This is repeated multiple times, you can pull this out to a public `ToggleInput` void that takes the input flag as a param